types/src/lib.rs

```diff
@@ -44,6 +44,95 @@ pub mod response;
 pub mod error;
 
 pub use error::{ErrorCode, ErrorObject, ErrorObjectOwned};
+pub use http::Extensions;
 pub use params::{Id, InvalidRequestId, Params, ParamsSequence, SubscriptionId, TwoPointZero};
-pub use request::{InvalidRequest, Notification, NotificationSer, Request, RequestSer};
+pub use request::{InvalidRequest, Notification, Request};
 pub use response::{Response, ResponsePayload, SubscriptionPayload, SubscriptionResponse, Success as ResponseSuccess};
+
+/// Deserialize calls, notifications and responses with HTTP extensions.
+pub mod deserialize_with_ext {
```

Comment:
These only appear to be used in one place, so I wonder if it's worth just having the logic there rather than exposing it form this crate (which makes it a public API)? Maybe it's useful for others though and not just an internal thing?